client/signup/steps/site-or-domain/index.jsx

```diff
@@ -171,6 +182,32 @@ class SiteOrDomain extends Component {
 	};
 
 	render() {
+		const { translate } = this.props;
+		const productsLoaded = ! isEmpty( this.props.productsList );
+
+		if ( productsLoaded && ! this.getDomainName() ) {
+			const headerText = translate( 'Unsupported domain.' );
+			const subHeaderText = translate(
+				'Please visit {{a}}wordpress.com/domains{{/a}} to search for another domain.',
+				{
+					components: {
+						a: <a href={ 'https://wordpress.com/domains' } />
+					}
+				}
+			);
+
+			return (
```

Comment:
Could you show something else than `StepWrapper`? It's very confusing - it says `Step 1 of 4`, but shows what's supposed to be an error message redirecting somewhere else. Maybe `EmptyContent` would be good instead?